.github/workflows/exhaustive.yml

```diff
@@ -0,0 +1,55 @@
+# Exhaustive Mode
+
+name: exhaustive
+
+on:
+  workflow_dispatch:
+    inputs:
+      git-ref:
+        description: 'Git Ref to Build (hash: 40 hex digits)'
+        required: true
+      # option:
+      #   description: use --override to allow any ref
+      #   required: false
+      #   default: ''
+jobs:
+  build:
+    name: Build for Production
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone Repository (Custom Ref)
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.git-ref }}
+          lfs: false
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('tools/**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Cache local npm repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+            node-
+      - name: Build with Maven
+        run: >
+          mvn -s .github/workflows/mvn-settings.xml -B compile install package --file tools/pom.xml -pl cldr-code
+          -DskipTests=true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Test with maven
+        run: >
+          mvn -s .github/workflows/mvn-settings.xml -B test --file tools/pom.xml
+          -pl cldr-code -Dorg.unicode.cldr.unittest.testArgs='-e10 -n'
```

Comment:
is "-e10" the main thing that's different here from the "build" part of production.yml?

it would be nice if the common parts could be encapsulated rather than duplicated -- but I'm not sure how to re-use code in workflows...

Reply:
This does fewer steps than the others. Yes we can try to refactor, but there are more divergent changes that will happen with this workflow.

Reply:
Workflows are extremely complicated to debug. I'd rather not prematurely refactor.